`dapr run -f` docs (#3093)
dapr/docs: daprdocs/content/en/developing-applications/local-development/multi-app-dapr-run.md

---
type: docs
title: Run multiple applications with one command
linkTitle: Multi-app Dapr Run
weight: 2000
description: Learn the scenarios around running multiple applications with one Dapr command
---

{{% alert title="Note" color="primary" %}}
 Multi-app `dapr run` is currently a preview feature only supported in Linux/MacOS. 
{{% /alert %}}

> Let's say you want to run several applications in local mode while being able to replication the production scenario. While in Kubernets mode, you'd be able to use helm/deployment YAML files, self-hosted mode required you to:

Reviewer: ```suggestion
Let's say you want to run several applications locally to test them together, similar to a production scenario. With a local Kubernetes cluster, you'd be able to use helm/deployment YAML files to do this, however this means building them as containers and setting up Kubernetes. Instead you simply want to run them as local executables in self-hosted mode.  However self-hosted mode requires you to:
```

- Run multiple `dapr run` commands

> - Keep track of all ports opened

Reviewer: ```suggestion
- Keep track of all ports opened (you cannot have duplicate ports for different applications)
```

- Remember the resources folders and configuration files that each application refers to
- Recall all of the additional flags you used to tweak the `dapr run` command behavior (`--app-health-check-path`, `--dapr-grpc-port`, `--unix-domain-socket`, etc.)

> With the Multi-app Dapr Run feature, you can easily start multiple Dapr applications in self-hosted mode using a single `dapr run -f` command.

Reviewer: ```suggestion
With Multi-app Run, you can easily start multiple applications in self-hosted mode using a single `dapr run -f myapps.yaml` command.
```

## How does it work?

> Currently, upon running [`dapr init`]({{< ref install-dapr-selfhost.md >}}), Dapr initializes in the `~/.dapr/` directory on Linux, where the default configurations and resources are stored.

Reviewer: ```suggestion
When running [`dapr init`]({{< ref install-dapr-selfhost.md >}}), this initializes a directory where the default configurations and resources are stored.
```

> For running multiple applications, `dapr init` will initialize the following `~/.dapr/` directory structure:

Reviewer: @hhunter-ms - Let's chat about the layout here. Not sure we should be trying explain dapr init. It is best to start from the mult-app run file and simply explain that for each application that you need to run. Then at the end you can say there is a simplified version that assumes this directory structure and so you can remove these optional values. Then show a simpler version of the same multi-app file. Make sense?
